migrate.py

```diff
@@ -649,6 +682,304 @@ def copy_tests(plugin, coll, spec, args):
+                if plugin_collection != collection:
+                    integration_tests_add_to_deps(collection, plugin_collection)
+            except LookupError:
+                new_plugin_names.append(plugin_name)
+
+        config.set(section, keyword, ','.join(new_plugin_names))
+
+
+def rewrite_yaml(src, dest, namespace, collection, spec):
+    contents = read_yaml_file(src)
+    _rewrite_yaml(contents, namespace, collection, spec)
+    write_yaml_into_file_as_is(dest, contents)
+
+
+def _rewrite_yaml(contents, namespace, collection, spec):
+    if isinstance(contents, list):
+        for el in contents:
+            _rewrite_yaml(el, namespace, collection, spec)
+    elif isinstance(contents, Mapping):
+        _rewrite_yaml_mapping(contents, namespace, collection, spec)
+
+
+KEYWORD_TO_PLUGIN_MAP = {
+    'ansible_become_method': 'become',
+    'ansible_connection': 'connection',
+    'ansible_shell_type': 'shell',
+    'become_method': 'become',
+    'cache_plugin': 'cache',
+    'connection': 'connection',
+    'plugin': 'inventory',
+    'strategy': 'strategy',
+}
+
+
+def _rewrite_yaml_mapping(el, namespace, collection, spec):
+    assert isinstance(el, Mapping)
+
+    _rewrite_yaml_mapping_keys(el, namespace, collection, spec)
+    _rewrite_yaml_mapping_values(el, namespace, collection, spec)
+
+
+manual_check = []
+def add_to_manual_check(key, value):
+    global manual_check
+    # FIXME add the file too
+    manual_check.append((key, value))
+
+
+def _rewrite_yaml_mapping_keys(el, namespace, collection, spec):
+    for key in el.keys():
+        if is_reserved_name(key):
+            logger.debug('skipping reserved name ' + key)
+            continue
+
+        plugin_type = KEYWORD_TO_PLUGIN_MAP.get(key)
+        if plugin_type:
+            try:
+                plugin_collection = get_plugin_collection(el[key], plugin_type, spec)
+                el[key] = get_plugin_fqcn(namespace, plugin_collection, el[key])
+                if plugin_collection != collection:
+                    integration_tests_add_to_deps(collection, plugin_collection)
+            except LookupError:
+                if '{{' in el[key]:
+                    add_to_manual_check(key, el[key])
+
+        prefix = 'with_'
+        if prefix in key:
+            prefix_len = len(prefix)
+
+            if not key.startswith(prefix):
+                continue
+            plugin_name = key[prefix_len:]
+            try:
+                plugin_collection = get_plugin_collection(plugin_name, 'lookup', spec)
+                el[prefix + get_plugin_fqcn(namespace, plugin_collection, plugin_name)] = el[key]
+                del el[key]
+                if plugin_collection != collection:
+                    integration_tests_add_to_deps(collection, plugin_collection)
+            except LookupError:
+                pass
+
+        try:
+            for coll in spec.keys():
+                for module in get_plugins_from_collection(coll, 'modules', spec):
```

Comment:
You should probably wrap this function call with try/except. using it around the loop shadows other exceptions. It's important because you work with index-based access a lot here.

Reply:
Good point. That was a result from moving code around. Fixed.